lib/improver/nbhood_percentiles.py

```diff
@@ -0,0 +1,375 @@
+            radius = self._find_radii(num_ens)
+            ranges = max(Utilities.get_neighbourhood_width_in_grid_cells(
+                cube, radius, MAX_RADIUS_IN_GRID_CELLS))
+            cube_new = self.method.run(cube, ranges)
+        else:
+            cube_lead_times = (
+                Utilities.find_required_lead_times(cube))
+            # Interpolate to find the radius at each required lead time.
+            required_radii = (
+                self._find_radii(num_ens,
+                                  cube_lead_times=cube_lead_times))
+
+            cubes = iris.cube.CubeList([])
+            for cube_slice, radius in (
+                    zip(cube.slices_over("time"),
+                        required_radii)):
+                ranges = max(Utilities.get_neighbourhood_width_in_grid_cells(
+                    cube, radius, MAX_RADIUS_IN_GRID_CELLS))
+                cube_perc = self.method.run(cube_slice, ranges)
+                cube_perc = iris.util.new_axis(cube_perc, "time")
+                cubes.append(cube_perc)
+                cube_new = concatenate_cubes(cubes,
+                                             coords_to_slice_over=["time"])
+
+        return cube_new
+
+class CircularKernelNumpy(object):
+    """
+    Methods for use in calculating percentiles from a 2D circular neighbourhood.
+
+    A maximum kernel radius of 500 grid cells is imposed in order to
+    avoid computational ineffiency and possible memory errors.
+    """
+    def __init__(self,
+                 percentiles=PercentileConverter.DEFAULT_PERCENTILES):
+        """
+
+        Parameters
+        ----------
+
+        percentiles : list (optional)
+            Percentile values at which to calculate; if not provided uses
+            DEFAULT_PERCENTILES from percentile module.
+        """
+        self.percentiles = percentiles
+
+    def __repr__(self):
+        """Represent the configured class instance as a string."""
+        result = ('<CircularKernelNumpy: percentiles: {}>')
+        return result.format(self.percentiles)
+
+    def run(self, cube, ranges):
+        """
+        Method to apply a circular kernel to the data within the input cube in
+        order to derive percentiles over the kernel.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube containing array to apply processing to.
+        ranges : Int
+            Number of grid cells in the x and y direction used to create
+            the kernel.
+
+        Returns
+        -------
+        outcube : Iris.cube.Cube
+            Cube containing the percentile fields.
+            Has percentile as an added dimension.
+
+        """
+        # Take data array and identify X and Y axes indices
+        try:
+            for coord_name in ['projection_x_coordinate',
+                               'projection_y_coordinate']:
+                cube.coord(coord_name)
+        except CoordinateNotFoundError:
```

Comment:
What is going to happen when we try apply neighbourhood methods to Global data which will have lat / lon